File: 1806-count-of-matches-in-tournament/1806-count-of-matches-in-tournament.py
class Solution:
    def numberOfMatches(self, n: int) -> int:
        if(n==1):
            return 0
        if(n==2):
            return 1
        c=1
        while(n!=2):
            if(n%2==0):
                c+=(n-(n//2))
                n=(n-(n//2))
            else:
                c+=((n-1)//2)
                r=(n-1)//2+1
                n=r
        return c

        

File: 1806-count-of-matches-in-tournament/1806-count-of-matches-in-tournament.cpp
class Solution {
public:
    int numberOfMatches(int n) {
        int c=0;
        if(n==1){
            return 0;
        }
        while(n>1){
            if(n%2==0){
                c+=n-(n/2);
                n=n-(n/2);

            }
            else{
                c+=(n - 1) / 2;
                n=(n - 1) / 2 + 1;
            }
        }
        return c;
        
    }
};